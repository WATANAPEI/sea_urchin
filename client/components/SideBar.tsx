import React from "react";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";
import { BrowserRouter as Router, Route, NavLink } from "react-router-dom";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    sideBar: {
      color: "red",
      margin: "0 5px",
      height: "100%"
    },
    li: {
      listStyle: "none"
    },
    navLink: {
      width: "100%",
      textDecorationLine: "none"
    }
  })
);

function SideBar() {
  const classes = useStyles();
  return (
    <Paper className={classes.sideBar}>
      <MenuList className={classes.li}>
        <MenuItem>
          <NavLink to="/lang/words/" className={classes.navLink}>
            Word Card
          </NavLink>
        </MenuItem>
        <MenuItem>
          <NavLink to="/lang/words/wordlist" className={classes.navLink}>
            Word List
          </NavLink>
        </MenuItem>
      </MenuList>
    </Paper>
  );
}

export default SideBar;
