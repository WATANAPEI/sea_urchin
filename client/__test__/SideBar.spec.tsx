import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
// @ts-ignore
import SideBar from "../components/SideBar.tsx";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";
import { NavLink } from "react-router-dom";

describe("</SideBar />", () => {
  it("has a MenuList and MenuItems", () => {
    const wrapper = createShallow()(<SideBar />);
    expect(wrapper.find(MenuList)).toHaveLength(1);
    expect(wrapper.find(MenuItem).exists()).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders NavLink with text", () => {
    const testText = "sideBar text";
    const wrapper = createShallow()(<SideBar />);
//    console.log(wrapper.text());
    expect(wrapper.find(NavLink).exists()).toEqual(true);
  });
});
