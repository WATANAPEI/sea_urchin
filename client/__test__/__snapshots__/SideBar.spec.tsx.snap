// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</SideBar /> has a MenuList and MenuItems 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-sideBar-1"
>
  <ForwardRef(MenuList)
    className="makeStyles-li-2"
  >
    <WithStyles(ForwardRef(MenuItem))>
      <NavLink
        className="makeStyles-navLink-3"
        to="/lang/words/"
      >
        Word Card
      </NavLink>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))>
      <NavLink
        className="makeStyles-navLink-3"
        to="/lang/words/wordlist"
      >
        Word List
      </NavLink>
    </WithStyles(ForwardRef(MenuItem))>
  </ForwardRef(MenuList)>
</WithStyles(ForwardRef(Paper))>
`;
